src/message_handler.py

```diff
@@ -62,8 +62,9 @@ def handle_incoming_message_core(ctx:Context, event, in_flight):
     start = time.time()
     parsed_event = json.loads(event)
     ctx.log(parsed_event)
-    messenger = messenger_factory.messenger_by_type[parsed_event["source"]]
-    
+    chat_id = str(parsed_event["event"]['message']['chat']['id'])
```

Comment:
This code would work for Telegram, but not WhatsApp.

Note how chat_id is set in wa.py; ideally, add another variant of messenger_factory.make_messenger(raw_message) or perhaps messenger_factory.make_messenger(source, message).

There you can handle this correctly per source type (tg/wa).
As a side effect will also make this nicer, one-liner code.

Reply:
yep, i missed that.
i've committed a fix, even though it's not as concise as i'd want it to be... whatsapp's event kind complicates things a bit.